### ISSUE TITLE
Install-PSResource fails to install dependencies

### ISSUE THREAD

**Issue by ThomasNieto** [PowerShell/PSResourceGet#664]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PowerShellGet)
- [X] Search the existing issues.

### Steps to reproduce

I am trying to install the winget module and it fails when processing package management even though its already installed.

### Expected behavior

```console
Installed successfully.
```

### Actual behavior

```console
C:\> Install-PSResource winget
WARNING: Resource 'PackageManagement' with version '1.4.7.0' is already installed.  If you would like to reinstall, please run the cmdlet again with the -Reinstall parameter
Install-PSResource: Unable to successfully install package 'WinGet': 'Cannot generate a PowerShell object for a ScriptBlock evaluating dynamic expressions. Dynamic expression: @{
        RootModule = 'WinGet.psm1'
        ModuleVersion = '0.0.8'
        GUID = '468ef37a-2557-4c10-92ec-783ec1e41639'
        Author = 'Ethan Bergstrom'
        Copyright = ''
        Description = 'Package Management (OneGet) provider that facilitates installing WinGet packages from any NuGet repository.'
        # Refuse to load in CoreCLR if PowerShell below 7.0.1 due to regressions with how 7.0 loads PackageManagement DLLs
        # https://github.com/PowerShell/PowerShell/pull/12203
        PowerShellVersion = if ($PSEdition -eq 'Core') {
                '7.0.1'
        } else {
                '5.1'
        }
        RequiredModules = @(
                @{
                        ModuleName='PackageManagement'
                        ModuleVersion='1.1.7.2'
                },
                @{
                        ModuleName='Cobalt'
                        ModuleVersion='0.2.0'
                }
        )
        PrivateData = @{
                PackageManagementProviders = 'WinGet.psm1'
                PSData = @{
                        # Tags applied to this module to indicate this is a PackageManagement Provider.
                        Tags = @('PackageManagement','Provider','WinGet','PSEdition_Desktop','PSEdition_Core','Windows')
                        # A URL to the license for this module.
                        LicenseUri = 'https://github.com/ethanbergstrom/WinGet/blob/current/LICENSE.txt'
                        # A URL to the main website for this project.
                        ProjectUri = 'https://github.com/ethanbergstrom/WinGet'
                        # ReleaseNotes of this module
                        ReleaseNotes = 'Please see https://github.com/ethanbergstrom/winget/blob/master/CHANGELOG.md for release notes'
                }
        }
}.'
```

### Error details

```console
Exception             :
    Type           : System.Management.Automation.PSInvalidOperationException
    ErrorRecord    :
        Exception             :
            Type    : System.Management.Automation.ParentContainsErrorRecordException
            Message : Unable to successfully install package 'WinGet': 'Cannot generate a PowerShell object for a ScriptBlock evaluating dynamic expressions. Dynamic expression: @{
        RootModule = 'WinGet.psm1'
        ModuleVersion = '0.0.8'
        GUID = '468ef37a-2557-4c10-92ec-783ec1e41639'
        Author = 'Ethan Bergstrom'
        Copyright = ''
        Description = 'Package Management (OneGet) provider that facilitates installing WinGet packages from any NuGet repository.'
        # Refuse to load in CoreCLR if PowerShell below 7.0.1 due to regressions with how 7.0 loads PackageManagement DLLs
        # https://github.com/PowerShell/PowerShell/pull/12203
        PowerShellVersion = if ($PSEdition -eq 'Core') {
                '7.0.1'
        } else {
                '5.1'
        }
        RequiredModules = @(
                @{
                        ModuleName='PackageManagement'
                        ModuleVersion='1.1.7.2'
                },
                @{
                        ModuleName='Cobalt'
                        ModuleVersion='0.2.0'
                }
        )
        PrivateData = @{
                PackageManagementProviders = 'WinGet.psm1'
                PSData = @{
                        # Tags applied to this module to indicate this is a PackageManagement Provider.
                        Tags = @('PackageManagement','Provider','WinGet','PSEdition_Desktop','PSEdition_Core','Windows')

                        # A URL to the license for this module.
                        LicenseUri = 'https://github.com/ethanbergstrom/WinGet/blob/current/LICENSE.txt'

                        # A URL to the main website for this project.
                        ProjectUri = 'https://github.com/ethanbergstrom/WinGet'

                        # ReleaseNotes of this module
                        ReleaseNotes = 'Please see https://github.com/ethanbergstrom/winget/blob/master/CHANGELOG.md for release notes'
                }
        }
}.'
            HResult : -2146233087
        CategoryInfo          : InvalidOperation: (:) [], ParentContainsErrorRecordException
        FullyQualifiedErrorId : InvalidOperation
    Message        : Unable to successfully install package 'WinGet': 'Cannot generate a PowerShell object for a ScriptBlock evaluating dynamic expressions. Dynamic expression: @{
        RootModule = 'WinGet.psm1'
        ModuleVersion = '0.0.8'
        GUID = '468ef37a-2557-4c10-92ec-783ec1e41639'
        Author = 'Ethan Bergstrom'
        Copyright = ''
        Description = 'Package Management (OneGet) provider that facilitates installing WinGet packages from any NuGet repository.'
        # Refuse to load in CoreCLR if PowerShell below 7.0.1 due to regressions with how 7.0 loads PackageManagement DLLs
        # https://github.com/PowerShell/PowerShell/pull/12203
        PowerShellVersion = if ($PSEdition -eq 'Core') {
                '7.0.1'
        } else {
                '5.1'
        }
        RequiredModules = @(
                @{
                        ModuleName='PackageManagement'
                        ModuleVersion='1.1.7.2'
                },
                @{
                        ModuleName='Cobalt'
                        ModuleVersion='0.2.0'
                }
        )
        PrivateData = @{
                PackageManagementProviders = 'WinGet.psm1'
                PSData = @{
                        # Tags applied to this module to indicate this is a PackageManagement Provider.
                        Tags = @('PackageManagement','Provider','WinGet','PSEdition_Desktop','PSEdition_Core','Windows')

                        # A URL to the license for this module.
                        LicenseUri = 'https://github.com/ethanbergstrom/WinGet/blob/current/LICENSE.txt'

                        # A URL to the main website for this project.
                        ProjectUri = 'https://github.com/ethanbergstrom/WinGet'

                        # ReleaseNotes of this module
                        ReleaseNotes = 'Please see https://github.com/ethanbergstrom/winget/blob/master/CHANGELOG.md for release notes'
                }
        }
}.'
    InnerException :
        Type       : System.InvalidOperationException
        TargetSite :
            Name          : SafeGetValue
            DeclaringType : System.Management.Automation.Language.Ast
            MemberType    : Method
            Module        : System.Management.Automation.dll
        Message    : Cannot generate a PowerShell object for a ScriptBlock evaluating dynamic expressions. Dynamic expression: @{
        RootModule = 'WinGet.psm1'
        ModuleVersion = '0.0.8'
        GUID = '468ef37a-2557-4c10-92ec-783ec1e41639'
        Author = 'Ethan Bergstrom'
        Copyright = ''
        Description = 'Package Management (OneGet) provider that facilitates installing WinGet packages from any NuGet repository.'
        # Refuse to load in CoreCLR if PowerShell below 7.0.1 due to regressions with how 7.0 loads PackageManagement DLLs
        # https://github.com/PowerShell/PowerShell/pull/12203
        PowerShellVersion = if ($PSEdition -eq 'Core') {
                '7.0.1'
        } else {
                '5.1'
        }
        RequiredModules = @(
                @{
                        ModuleName='PackageManagement'
                        ModuleVersion='1.1.7.2'
                },
                @{
                        ModuleName='Cobalt'
                        ModuleVersion='0.2.0'
                }
        )
        PrivateData = @{
                PackageManagementProviders = 'WinGet.psm1'
                PSData = @{
                        # Tags applied to this module to indicate this is a PackageManagement Provider.
                        Tags = @('PackageManagement','Provider','WinGet','PSEdition_Desktop','PSEdition_Core','Windows')

                        # A URL to the license for this module.
                        LicenseUri = 'https://github.com/ethanbergstrom/WinGet/blob/current/LICENSE.txt'

                        # A URL to the main website for this project.
                        ProjectUri = 'https://github.com/ethanbergstrom/WinGet'

                        # ReleaseNotes of this module
                        ReleaseNotes = 'Please see https://github.com/ethanbergstrom/winget/blob/master/CHANGELOG.md for release notes'
                }
        }
}.
        Source     : System.Management.Automation
        HResult    : -2146233079
        StackTrace :
   at System.Management.Automation.Language.Ast.SafeGetValue(Boolean skipHashtableSizeCheck)
   at System.Management.Automation.Language.Ast.SafeGetValue()
   at Microsoft.PowerShell.PowerShellGet.UtilClasses.Utils.TryParseModuleManifest(String moduleFileInfo, PSCmdlet cmdletPassedIn, Hashtable& parsedMetadataHashtable) in D:\a\_work\1\s\src\code\Utils.cs:line
470
   at Microsoft.PowerShell.PowerShellGet.Cmdlets.InstallHelper.InstallPackage(IEnumerable`1 pkgsToInstall, String repoUrl, PSCredential credential, Boolean isLocalRepo) in
D:\a\_work\1\s\src\code\InstallHelper.cs:line 494
    HResult        : -2146233079
TargetObject          : Microsoft.PowerShell.PowerShellGet.Cmdlets.InstallPSResource
CategoryInfo          : InvalidOperation: (Microsoft.PowerShel…s.InstallPSResource:InstallPSResource) [Install-PSResource], PSInvalidOperationException
FullyQualifiedErrorId : InstallPackageFailed,Microsoft.PowerShell.PowerShellGet.Cmdlets.InstallPSResource
InvocationInfo        :
    MyCommand        : Install-PSResource
    ScriptLineNumber : 1
    OffsetInLine     : 1
    HistoryId        : 15
    Line             : Install-PSResource winget
    PositionMessage  : At line:1 char:1
                       + Install-PSResource winget
                       + ~~~~~~~~~~~~~~~~~~~~~~~~~
    InvocationName   : Install-PSResource
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo :
```

### Environment data

```powershell
ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Binary     3.0.12     beta       PowerShellGet                       {Find-PSResource, Get-PSResource, Get-PSResourceRepository, Install-PSResource…}

Name                           Value
----                           -----
PSVersion                      7.2.4
PSEdition                      Core
GitCommitId                    7.2.4
OS                             Microsoft Windows 10.0.19043
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

Name              Version Prerelease Repository Description
----              ------- ---------- ---------- -----------
Cobalt            0.3.1.0            PSGallery  A PowerShell Crescendo wrapper for WinGet
PackageManagement 1.4.7.0            PSGallery  PackageManagement (a.k.a. OneGet) is a new way to discover and install software packages from around the web.…
```

**Comment by SydneyhSmith** [comment]
Thanks looks like this is related to https://github.com/PowerShell/PowerShellGet/issues/533

**Comment by alerickson** [comment]
Resolved with PR #681
